File: ttknp-ecommerce-backend/src/main/java/com/ttknpdev/backend/entities/OrderItem.java
package com.ttknpdev.backend.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "order_items")
@Getter
@Setter
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long oiId;
    private String imageUrl;
    private BigDecimal price;
    private Integer quantity;
    private Long productPid; // fk mapped
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "order_oid")
    private Order order; // fk
}

File: ttknp-ecommerce-fontend/src/app/entities/order-item.ts
import {CartItem} from "./cart-item";

export class OrderItem {

  public imageUrl: String
  public quantity: number
  public price: number
  public productPid: number

  constructor(cartItem: CartItem) {
    this.imageUrl = cartItem.imageUrl;
    this.quantity = cartItem.quantity;
    this.price = cartItem.price;
    this.productPid = cartItem.pid;
  }
}
